### PR TITLE
Add support for gz-compressed gfa1 and gfa2 files

### PR DESCRIPTION
**Pull request by gallardoalba**
This PR includes gzip compression option in gfa1 datasets.

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [x] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

Compress a gfa1 dataset.

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: lib/galaxy/config/sample/datatypes_conf.xml.sample [hunk 1]
```diff
@@ -28,8 +28,8 @@
     <datatype extension="anvio_variability" type="galaxy.datatypes.tabular:TSV" display_in_upload="false" subclass="true" />
     <datatype extension="arff" type="galaxy.datatypes.text:Arff" mimetype="text/plain" display_in_upload="true"/>
     <datatype extension="paf" auto_compressed_types="gz" type="galaxy.datatypes.text:Paf" mimetype="text/plain" display_in_upload="true"/>
-    <datatype extension="gfa1" type="galaxy.datatypes.text:Gfa1" mimetype="text/plain" display_in_upload="true"/>
-    <datatype extension="gfa2" type="galaxy.datatypes.text:Gfa2" mimetype="text/plain" display_in_upload="true"/>
+    <datatype extension="gfa1" auto_compressed_types="gz" type="galaxy.datatypes.text:Gfa1" mimetype="text/plain" display_in_upload="true"/>
+    <datatype extension="gfa2" auto_compressed_types="gz" type="galaxy.datatypes.text:Gfa2" mimetype="text/plain" display_in_upload="true"/>
```
Comment (mvdbeek) [inline lib/galaxy/config/sample/datatypes_conf.xml.sample:32]: Do tools that consume gfa files expect them to be tabix-indexable if they're gz-compressed (IOW do we need to use bgzip) ? Can you also add a compressed copy of the two gfa files in lib/galaxy/datatypes/test ?

Comment (gallardoalba) [reply lib/galaxy/config/sample/datatypes_conf.xml.sample:32]: Sure, thanks for reviewing. I'll ask about the indexation.

Comment (gallardoalba) [reply lib/galaxy/config/sample/datatypes_conf.xml.sample:32]: I asked the developer of gfastats and he told me that it is not necessary to be tabix-indexable.

**Review by mvdbeek — APPROVED** [review]
Do you want to target 22.01 ?

**Comment by gallardoalba** [comment]
> Do you want to target 22.01?

Yes, it would be nice.

**Pull request closed without merging**